typing/typecore.ml

```diff
@@ -5582,13 +5587,30 @@ let report_error ~loc env = function
            fprintf ppf "This expression has type")
         (function ppf ->
            fprintf ppf "but an expression was expected of type");
-  | Apply_non_function typ ->
+  | Apply_non_function (typ, arg_loc, prev_arg_loc) ->
       begin match get_desc typ with
         Tarrow _ ->
-          Location.errorf ~loc
+          let app_loc =
+            (* Span the application, including the extra argument. *)
+            { Location.loc_start = loc.Location.loc_start;
+              loc_end = arg_loc.loc_end;
+              loc_ghost = false }
+          and tail_loc =
+            (* Possible location for a ';'. *)
+            let loc_start = prev_arg_loc.loc_end in
+            { Location.loc_start;
+              loc_end = { loc_start with pos_cnum = loc_start.pos_cnum + 1 };
+              loc_ghost = false }
+          in
+          let sub = [
+            Location.msg ~loc:tail_loc "Maybe you forgot a ';'.";
```

Comment:
Since this is a suggestion that might be wrong, I would like to have a `Hint` here (seen as "a suggestion that might be wrong") which ends with a question mark:
```
Hint: Did you forget a ';'?
```